src/requests/create-signature.ts

```diff
@@ -0,0 +1,112 @@
+import * as crypto from 'crypto'
+import type {
+  CanonicalRequest,
+  NormalizedCanonicalRequest,
+  NormalizedHeaders,
+  Secret,
+  Timestamp,
+} from './typings'
+import { CanonicalRequestValidator, SecretValidator, TimestampValidator } from './typings'
+import { getNormalizedEncodedURI, getNormalizedHeaders } from './utils'
+import { ContentfulSigningHeader } from './constants'
+
+const hash = (normalizedCanonicalRequest: NormalizedCanonicalRequest, secret: string) => {
+  const stringifiedHeaders = normalizedCanonicalRequest
+    .headers!.map(([key, value]) => `${key}:${value}`)
+    .join(';')
+
+  const stringifiedRequest = [
+    normalizedCanonicalRequest.method,
+    normalizedCanonicalRequest.path,
+    stringifiedHeaders,
+    normalizedCanonicalRequest.body,
+  ].join('\n')
+
+  const hmac = crypto.createHmac('sha256', secret)
+
+  hmac.update(stringifiedRequest)
+
+  return hmac.digest('hex')
+}
+
+const enrichNormalizedHeadersWithMetadata = (headers: NormalizedHeaders, timestamp: number) => {
+  const result = [...headers]
+  const headerKeys = headers.map(([key]) => key)
+  const joinedSignedHeaders = headerKeys
+    // We always sign timestamp
+    .concat(ContentfulSigningHeader.Timestamp, ContentfulSigningHeader.SignedHeaders)
+    .join(',')
+
+  result.push([ContentfulSigningHeader.Timestamp, timestamp.toString()])
+  result.push([ContentfulSigningHeader.SignedHeaders, joinedSignedHeaders])
```

Comment:
What if `headers` will already have timestamp/signed headers? As it will when verifying a request?

Reply:
You're right, this is a leftover from before when the timestamp and signed-headers weren't in the list. Will fix it. Thanks for catching it